android/src/main/java/com/oney/WebRTCModule/UvcEnumerator.java

```diff
@@ -0,0 +1,73 @@
+package com.oney.WebRTCModule;
+
+import android.content.Context;
+import android.hardware.Camera;
+import android.hardware.usb.UsbDevice;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
+import com.jiangdg.ausbc.camera.CameraUvcStrategy;
+
+import org.webrtc.SurfaceViewRenderer;
+
+import java.util.List;
+
+public class UvcEnumerator {
+
+    private static final String TAG = WebRTCModule.TAG;
+
+    CameraUvcStrategy mUvcStrategy; // From Jiang Dongguo's AUSBC library
+
+    public UvcEnumerator(CameraUvcStrategy cameraUvcStrategy) {
+        try {
+            mUvcStrategy = cameraUvcStrategy;
+        } catch (Throwable tr) {
+            Log.e(TAG, "Error on UvcEnumerator.UvcEnumerator", tr);
+        }
+    }
+
+    ReadableArray enumerateDevices() {
+        WritableArray array = Arguments.createArray();
+
+        try {
+            List<UsbDevice> usbDevices = mUvcStrategy.getUsbDeviceList(null);
+
+            if (usbDevices != null) {
+
+                if (!usbDevices.isEmpty()) {
+
+                    for (int i = 0; i < usbDevices.size(); i++) {
+                        UsbDevice dev = usbDevices.get(i);
+
+                        WritableMap params = Arguments.createMap();
+
+                        params.putString("facing", "uvc");
+                        params.putString("deviceId", "uvc-" + dev.getDeviceId());
```

Comment:
I used the uvc prefix on the deviceId to be able to identify when looking for a uvc device in the CameraController.

Reply:
There in UvcCapturer I remove the uvc prefix.

Reply:
Good idea!